### PR TITLE
Issue/11799 show announcement on app upgrade

### LINKED ISSUE #11799: What's new - show feature announcement on app upgrade
We should show the feature announcement when user upgrades the app.

Details of the logic are TBD, but current options are:

- Targeted announcement - each announcement will be targeted towards specific build code, and will only be displayed to you when your app version matches one in the announcement. You might not see any announcements at all if you skip versions in which we have them when upgrading.

- Most recent announcement - most recent announcement will always be displayed on version upgrade (only once). It’s still possible to skip some announcements, and an announcement you see might not be relevant (eg. You are upgrading from version 1 to 4. In version 2 we added some fancy features and release an announcement. In version 3 we changed or removed the fancy feature and had no announcement, so the user will see the announcement from v2 while on v4).

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #11799

This PR adds logic for displaying feature announcements on the app upgrade.

We decided to show announcements independent of app version - eg. if you are on v4 of the app, and there is an announcement in v5, you will see even if you skip v5 and upgrade straight to v6 or v7 (until we remove or replace announcement manually).

To test (using wasabi build):
- If you install this branch on top of develop - the announcement should be displayed when app starts.
- If you perform the fresh installation, an announcement will not be displayed.
- To trigger an announcement bump the [versionCode](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/11799-show-announcement-on-app-upgrade/WordPress/src/main/java/org/wordpress/android/ui/whatsnew/FeatureAnnouncementProvider.kt#L8) number and restart the app.
- Make sure the announcement is not shown in vanilla version.

PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by oguzkocer** [comment]
Hi there! `14.8` is being frozen, so this PR is being bumped to `14.9`. If these changes _have to_ be shipped with `14.8`, please target `release/14.8` and cc me in the PR so I can cut a new beta release. Also if you can keep the release dates in mind and bump the PRs & issues you are working on before the code freeze date, we'd really appreciate it.

**Extra notes:**
@khaykov There are a couple issues I want to note in this PR:
* Can you make sure to add a reviewer for the PR when you create it?
* Since this PR is opened on a Saturday before a code freeze, we expect one of 2 things to happen. Either ping platform9 and tell us that this PR needs to be included in the upcoming code freeze, in which case it really should have a reviewer and a notice that it's urgent. Or, target the next milestone. In the current state of the PR, it's not clear what the intention for the PR is, so we have to assume that it's OK to bump to the next release. Hope that makes sense!

Let me know if you have any questions!

**Comment by khaykov** [comment]
@oguzkocer Got it! Sorry, I totally missed that the code freeze is right around the corner for 14.8.

File: WordPress/src/main/java/org/wordpress/android/ui/whatsnew/FeatureAnnouncementProvider.kt [hunk 1]
```diff
@@ -27,7 +27,11 @@ class FeatureAnnouncementProvider @Inject constructor() {
     )
     )
 
-    fun getLatestFeatureAnnouncement(): FeatureAnnouncement {
+    fun getLatestFeatureAnnouncement(): FeatureAnnouncement? {
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/whatsnew/FeatureAnnouncementProvider.kt:30]: I added the `getFeatureAnnouncements(): List<FeatureAnnouncement>` method in upcoming PR, but it's not really solving the nullability of the announcement, it just moves the check upstream.

File: WordPress/src/main/java/org/wordpress/android/util/BuildConfigWrapper.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.wordpress.android.BuildConfig
+import javax.inject.Inject
+
+class BuildConfigWrapper @Inject constructor() {
+    fun getAppVersionCode(): Int {
+        return BuildConfig.VERSION_CODE
+    }
+
+    fun isFeatureAnnouncementEnabled(): Boolean {
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/util/BuildConfigWrapper.kt:11]: I added the check to AppSettingsFragment in upcoming PR :)

**Comment by khaykov** [comment]
Updated the logic around app versions. Hope I didn't mess up anything this time :)

The announcement should not be shown:
- Fresh app install.
- Updating from develop branch to this one without version code change.
- If the announcement with the same version has been already shown.

The announcement should be shown:
- When upgrading app (even when you skip some versions) and the announcement was not shown before.

**Pull request merged**